nx/lib/nx/defn/kernel.ex

```diff
@@ -536,23 +536,19 @@ defmodule Nx.Defn.Kernel do
   def not tensor when is_number(tensor), do: logical_not(tensor)
   def not tensor, do: Nx.logical_not(tensor)
 
-  defp logical_and(l, r) when Kernel.==(l, 0), do: zero(l, r)
-  defp logical_and(l, r) when Kernel.==(r, 0), do: zero(l, r)
-  defp logical_and(l, r), do: one(l, r)
+  defp logical_and(l, _) when Kernel.==(l, 0), do: zero()
```

Comment:
```suggestion
  defp logical_and(0, _), do: zero()
```
Why not hard-match like this?

Reply:
We need to handle both floats and integers and a match is type specific. :)

Reply:
Nice. == vs ===